### PR TITLE
Indentifier types: initial implementation

### PR DESCRIPTION
**Pull request by DFoxinator**
@be-dmitry 
@ingluisjimenez

### CODE REVIEW

File: src/Rules/AnonymousPercentageRule.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Behance\NBD\Gatekeeper\Rules;
+
+use Behance\NBD\Gatekeeper\Exceptions\ParameterValidationException;
+
+class AnonymousPercentageRule extends PercentageRuleAbstract {
```
Comment (ingluisjimenez) [inline src/Rules/AnonymousPercentageRule.php]: I thought these changes were going to happen on Core, not at this level

Comment (DFoxinator) [inline src/Rules/AnonymousPercentageRule.php]: No, we decided in here for now.

File: src/Rules/AuthenticatedPercentageRule.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Behance\NBD\Gatekeeper\Rules;
+
+use Behance\NBD\Gatekeeper\Exceptions\ParameterValidationException;
+
+class AuthenticatedPercentageRule extends PercentageRuleAbstract {
+
+  const RULE_NAME       = 'authenticated_percentage';
+  const IDENTIFIER_TYPE = 'authenticated';
```
Comment (ingluisjimenez) [inline src/Rules/AuthenticatedPercentageRule.php]: `RuleAbstract::IDENTIFIER_AUTHENTICATED`

File: src/Rules/AnonymousPercentageRule.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+  const RULE_NAME       = 'anonymous_percentage';
+  const IDENTIFIER_TYPE = 'anonymous';
```
Comment (ingluisjimenez) [inline src/Rules/AnonymousPercentageRule.php]: `RuleAbstract::IDENTIFIER_ANONYMOUS`

File: tests/unit/Rules/AnonymousPercentageRuleTest.php [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+namespace Behance\NBD\Gatekeeper\Rules;
+
+use Behance\NBD\Gatekeeper\Exceptions\InvalidIdentifierException;
+use Behance\NBD\Gatekeeper\Test\BaseTest;
+
+class AnonymousPercentageRuleTest extends BaseTest {
+
+  private $_rule;
+
+  private function _getRule( $percentage, $feature ) {
```
Comment (ingluisjimenez) [inline tests/unit/Rules/AnonymousPercentageRuleTest.php]: docblock, and move it to the bottom

**Comment by ingluisjimenez** [comment]
👍

File: src/RuleFactory.php [hunk 5]
```diff
@@ -55,13 +57,24 @@ public static function create( $type, array $params = [], $feature = null ) {
...
-        return new PercentageRule(
+        return new AuthenticatedPercentageRule(
+            self::_getRuleParam( 'percentage', $type, $params ),
+            $feature
+        );
+
+      case AnonymousPercentageRule::RULE_NAME:
+
+        if ( $feature == null ) {
+          throw new MissingRuleParameterException( 'Missing required $feature parameter for AnonymousPercentageRule' );
```
Comment (be-dmitry) [inline src/RuleFactory.php]: Non blocker: Wrap `$feature` in curly braces

**Comment by be-dmitry** [comment]
👍

**Pull request merged**